certification/internal/shell/base_on_ubi.go

```diff
@@ -12,10 +12,7 @@ type BaseOnUBICheck struct{}
 
 func (p *BaseOnUBICheck) Validate(image string, logger *logrus.Logger) (bool, error) {
```

Comment:
I like the idea of splitting the command execution from the output parsing. It however does not solve the public interface testing; I mean we are not still unit testing the `Validate()` method which is exposed externally. 

If there is no easy way to unit test `Validate()` method, I am good with testing the `checkBasedOnUbi` 👍

Reply:
+1 to @samira-barouti comment.  BaseOnUBICheck.checkBasedOnUbi makes the code less readable IMO.  Would simplifying the Validate function signature help?  Instead of passing in logger for example, we just import it 

`import (
  "os"
  log "github.com/sirupsen/logrus"
)`

Reply:
I am definitely in favor of removing logger. That would simplify that interface. There are some other tricks that can be employed to test the public interface. Let me take a stab at it.

Reply:
@bcrochet I'm missing it, what challenges arise from the logger being present in the signature?

Reply:
@komish The logger itself doesn't really introduce challenges, but it really doesn't belong IMHO. See #13 . The problem of the exec needed to be mocked is still a concern.

Reply:
Understood. Agreed, don't like it in the signature.

Reply:
logger piece handled in another PR. The Validate part should be handled now.